undertow/src/main/java/org/wildfly/extension/undertow/ActiveRequestTrackerDefinition.java

```diff
@@ -0,0 +1,112 @@
+package org.wildfly.extension.undertow;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import io.undertow.server.RequestStatistics;
+import io.undertow.server.handlers.ActiveRequestTrackerHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PersistentResourceDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.SimpleOperationDefinition;
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.capability.DynamicNameMappers;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceRegistry;
+
+public class ActiveRequestTrackerDefinition extends PersistentResourceDefinition {
+    static final RuntimeCapability<Void> ACTIVE_REQUEST_TRACKING_CAPABILITY =
+            RuntimeCapability.Builder.of(Capabilities.CAPABILITY_ACTIVE_REQUEST_TRACKING, true,
+                            ActiveRequestTrackerService.class)
+                    .setDynamicNameMapper(DynamicNameMappers.GRAND_PARENT)
+                    .build();
+    protected static final SimpleAttributeDefinition PREDICATE = new SimpleAttributeDefinitionBuilder(Constants.PREDICATE, ModelType.STRING, true)
+            .setAllowExpression(true)
+            .setValidator(PredicateValidator.INSTANCE)
+            .setRestartAllServices()
```

Comment:
I doubt that this attribute would require restarting *all* services. It should be sufficient to use Flag.RESTART_RESOURCE_SERVICES.

Reply:
Fixed locally.

Reply:
Perhaps it only needs to restart the services of the parent resource?